functions.sh

```diff
@@ -54,6 +54,7 @@ sonic_get_version() {
     local describe=$(git describe --tags)
     local latest_tag=$(git describe --tags --abbrev=0)
     local branch_name=$(git rev-parse --abbrev-ref HEAD)
```

Comment:
>branch_name [](http://example.com/codeflow?start=10&length=11)

Do you know why the old code does not work? Is the local folder under git control?

Reply:
When checking out code it does not set the local branch name, looks like the azure pipelines not support it very well.

Azure Pipelines:
vmss-soni00007E:/agent/_work/1/s$ git rev-parse --abbrev-ref HEAD
HEAD

Jenkins:
jenkins-worker-2:/data/johnar/workspace/broadcom/buildimage-brcm-all$ git branch -vv
* master 32b5b02 [sonic-platform-daemons] Update submodule (#7277)

Reply:
it is better to address this from the azp script. otherwise, we have to cherry-pick this pr all the way to 201811 branch.